app/controllers/MoreInformationController.scala

```diff
@@ -35,7 +36,19 @@ class MoreInformationController @Inject() (
   def onPageLoad(departureId: String): Action[AnyContent] = actions.requireData(departureId) {
     implicit request =>
       val lrn = request.userAnswers.lrn
-      Ok(view(lrn))
+      Ok(view(lrn, departureId))
   }
 
+  def onSubmit(departureId: String): Action[AnyContent] = actions.requireData(departureId) {
+    implicit request =>
+      val locationOfGoods = request.userAnswers.departureData.Consignment.LocationOfGoods
+      val isSimplified    = request.userAnswers.departureData.isSimplified
+
+      val nextPage = locationOfGoods match {
+        case None if !isSimplified => controllers.locationOfGoods.routes.LocationTypeController.onPageLoad(departureId, mode = NormalMode)
+        case None                  => controllers.locationOfGoods.routes.AuthorisationNumberController.onPageLoad(departureId, mode = NormalMode)
+      }
```

Comment:
Would it be possible to move this into the navigator. The ua should be available there to do the same logic